src/PostgREST/ApiRequest.hs

```diff
@@ -164,7 +165,13 @@ userApiRequest conf prefs req reqBody = do
 
 -- | Parses the Prefer header
 userPreferences :: AppConfig -> Request -> TimezoneNames -> Preferences.Preferences
-userPreferences conf req timezones = Preferences.fromHeaders (configDbTxAllowOverride conf) timezones $ requestHeaders req
+userPreferences conf req timezones = Preferences.fromHeaders isTxOverrideAllowed timezones $ requestHeaders req
+  where
+    isTxOverrideAllowed = case configDbTxEnd conf of
+      TxCommitAllowOverride   -> True
+      TxRollbackAllowOverride -> True
+      TxCommit                -> False
+      TxRollback              -> False
```

Comment:
This part also overall seems more verbose. Maybe at define it next to the types? Perhaps both these boolean results can be shortened somehow.